src/client/session/cluster_time.rs

```diff
@@ -10,11 +10,11 @@ use crate::bson::{Document, Timestamp};
 #[derive(Debug, Deserialize, Clone, Serialize, Derivative)]
 #[derivative(PartialEq, Eq)]
 #[serde(rename_all = "camelCase")]
-pub(crate) struct ClusterTime {
-    cluster_time: Timestamp,
+pub struct ClusterTime {
+    pub cluster_time: Timestamp,
```

Comment:
I don't think these should be public; this allows a user who owns a ClusterTime to modify the timestamp or the signature, which is not useful since it would presumably break it. If users do actually need to inspect these values, we should make them private and add getters, but I think that this is supposed to be a mostly opaque value to users (that they receive from the driver and then can pass back in other places without needing to care about the internals), in which case we wouldn't need getters. Since the struct derives Debug, users will still be able to log cluster times even if we don't add getters, so that wouldn't be a reason to add getters either.

Reply:
That makes sense to me, the reasoning for exposing these was mainly for inspecting, but I think deriving `Debug` should take care of that.